tests/unittests/GraphOptzTest.cpp

```diff
@@ -3871,3 +3871,82 @@ TEST_F(GraphOptz, SinkClipBelowReshape) {
   bindings_.get(in)->getHandle().randomize(-1.0, 1.0, mod_.getPRNG());
   checkNumericalEquivalence();
 }
+
+/// Test that Add after Deconv is folded into Bias add when the actual Add is
+/// is a broadcast of the bias.
+static void FoldDeconvolutionAddIntoBiasAdd(PlaceholderBindings &bindings,
+                                            Module &mod, Function *F, int RnL) {
+  dim_t batch = 2;
+  dim_t inC = 2;
+  dim_t outC = 5;
+  dim_t inH = 3;
+  dim_t inW = 3;
+  unsigned_t kernel = 3;
+  std::vector<uint32_t> pads = {0, 0, 0, 0};
+  std::vector<uint32_t> stride = {1, 1};
+
+  auto *input = mod.createPlaceholder(ElemKind::FloatTy, {2, inH, inW, inC},
+                                      "input", false);
+  auto *filter = mod.createConstant(ElemKind::FloatTy,
+                                    {outC, kernel, kernel, inC}, "filter");
+
+  auto *bias = mod.createConstant(ElemKind::FloatTy, {outC}, "bias");
+  bias->getPayloadMutable().getHandle<float>() = {1, 3, 5, 7, 9};
+
+  std::pair<dim_t, dim_t> outHW =
+      calculateDeconvOutputDims(inH, inW, {kernel, kernel}, stride, pads);
+  auto outTy = mod.uniqueType(ElemKind::FloatTy,
+                              {batch, outHW.first, outHW.second, outC});
+
+  ConvTransposeNode *DN =
+      F->createConvTranspose("Deconv", input, filter, bias, outTy,
+                             {kernel, kernel}, stride, {0, 0, 0, 0}, 1);
+
+  auto *CN = mod.createConstant(ElemKind::FloatTy,
+                                {batch, outHW.first, outHW.second, outC}, "c1");
+  auto *AN = RnL ? F->createAdd("add", DN, CN) : F->createAdd("add", CN, DN);
+
+  CN->getPayloadMutable().getHandle<float>() = {
+      1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3,
+      4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1,
+      2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4,
+      5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2,
+      3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5,
+      1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3,
+      4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1,
+      2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4,
+      5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2,
+      3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5,
+      1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5, 1, 2, 3, 4, 5};
+
+  SaveNode *S = F->createSave("save", AN);
+  bindings.allocate(S->getPlaceholder());
+
+  EXPECT_EQ(F->getNodes().size(), 3);
+  ::glow::optimize(F, CompilationMode::Infer);
+  EXPECT_EQ(F->getNodes().size(), 2);
+
+  {
+    ConvTransposeNode *DN = llvm::dyn_cast<ConvTransposeNode>(S->getInput());
+    EXPECT_TRUE(DN);
+
+    Constant *CN = llvm::dyn_cast<Constant>(DN->getBias());
+    EXPECT_TRUE(CN);
+
+    auto BH = CN->getPayloadMutable().getHandle();
+    EXPECT_EQ(BH.raw(0), 1 + 1);
```

Comment:
Should we check all of the values instead of the first 5?

Reply:
This is checking input bias only. For output I've added `checkNumericalEquivalence`